### PR TITLE
[ALSP] Implementation of Metrics Collection for Misbehavior Reporting in Reputation Management System (RMS)

### PR DESCRIPTION
**Pull request by yhassanzadeh13**
This PR introduces a key component of the Reputation Management System (RMS) within the Application Layer Spam Prevention (ALSP) framework for Flow blockchain. It specifically focuses on implementing the metrics collection logic necessary to process reported instances of misbehavior by the engine.

**Note**: To streamline the review and integration processes, the PR has been designed to be as concise and self-contained as possible. Given that the development of RMS represents a significant undertaking, we have chosen to break down its implementation into a series of smaller, self-sufficient PRs. This approach not only expedites the review and development processes but also enables us to gather valuable feedback early in the development stages.

### CODE REVIEW

**Review by peterargue — APPROVED** [review]

File: module/metrics/noop.go [hunk 1]
```diff
@@ -290,3 +290,6 @@ func (nc *NoopCollector) OnBehaviourPenaltyUpdated(f float64)
 func (nc *NoopCollector) OnIPColocationFactorUpdated(f float64)                            {}
 func (nc *NoopCollector) OnAppSpecificScoreUpdated(f float64)                              {}
 func (nc *NoopCollector) OnOverallPeerScoreUpdated(f float64)                              {}
+func (nc *NoopCollector) OnMisbehaviorReported(string, string) {
+	// no-p0
+}
```
Comment (kc1116) [inline module/metrics/noop.go:295]:
```suggestion
func (nc *NoopCollector) OnMisbehaviorReported(string, string) {}
```

File: network/alsp/manager_test.go [hunk 2]
```diff
@@ -12,6 +12,7 @@ import (
 
 	"github.com/onflow/flow-go/module/irrecoverable"
 	"github.com/onflow/flow-go/module/metrics"
+	mock2 "github.com/onflow/flow-go/module/mock"
```
Comment (kc1116) [inline network/alsp/manager_test.go:15]:
```suggestion
	mockmodule "github.com/onflow/flow-go/module/mock"
```

File: network/alsp/manager_test.go [hunk 3]
```diff
@@ -81,6 +83,56 @@ func TestHandleReportedMisbehavior(t *testing.T) {
 	unittest.RequireReturnsBefore(t, allReportsManaged.Wait, 100*time.Millisecond, "did not receive all reports")
 }
 
+// TestMisbehaviorReportMetrics tests the recording of misbehavior report metrics.
+// It checks that when a misbehavior report is received by the ALSP manager, the metrics are recorded.
+// It fails the test if the metrics are not recorded or if they are recorded incorrectly.
+func TestMisbehaviorReportMetrics(t *testing.T) {
+	alspMetrics := mock2.NewAlspMetrics(t)
```
Comment (kc1116) [inline network/alsp/manager_test.go:90]:
```suggestion
	alspMetrics := mockmodule.NewAlspMetrics(t)
```

**Review by kc1116 — APPROVED** [review]

File: network/alsp/manager_test.go (hunk 3 above)
Comment (yhassanzadeh13) [reply network/alsp/manager_test.go]: https://github.com/onflow/flow-go/pull/4268/commits/2778ee321fec83b876d914d3c7ac1ae68a434c3b

**Pull request merged**